File: component-core/src/main/java/com/plugin/component/IComponent.java
package com.plugin.component;
import android.app.Application;

import androidx.annotation.NonNull;

/**
 * 基础组件实现接口
 * Email yummyl.lau@gmail.com
 * Created by yummylau on 2018/01/25.
 */
public interface IComponent {

    void attachComponent(@NonNull Application application);

    void detachComponent();
}

File: component-core/src/main/java/com/plugin/component/IComponent.kt
package com.plugin.component

import android.app.Application

/**
 * 基础组件实现接口
 * Email yummyl.lau@gmail.com
 * Created by yummylau on 2018/01/25.
 */
interface IComponent {

    fun attachComponent(application: Application)

    fun detachComponent()
}
